Bloom.md

```diff
@@ -0,0 +1,341 @@
+    switching to Valkey (with ValkeyBloom).
+
+2. Users can generate an AOF file from a server (that has the ReBloom module loaded) and with an on-going bloom workload
+that creates bloom filters & inserts items into them. Next, this can be re-played on a Valkey Server (with ValkeyBloom loaded).
+Then, the user can move their existing bloom workload to the Valkey server (with ValkeyBloom loaded).
+
+Pros
+* Bloom objects can be created on the user's existing system (with Rebloom) and the user can also populate the objects
+    by adding items (BF.ADD/BF.MADD) **BEFORE** switching to Valkey (with ValkeyBloom).
+
+Cons
+* The user will need to manage AOF file generation on the server (with ReBloom) and ensure that the contents do not get
+    re-written (e.g. as a result of BGREWRITEAOF) into a RDB file. This is because the RDB file with bloom object data
+    generated by ReBloom is not compatible with ValkeyBloom. The main drawback here is that the AOF can exceed a size
+    limit and get truncated / re-written.
+
+### Memory Management
+
+The bloom data type also supports memory management related callbacks:
+
+* free: Frees the bloom filter object when it is deleted, expired or evicted.
+* defrag: Supports active defrag for Bloom filter objects
+* mem_usage: Reports bytes used by a Bloom object and is used by the MEMORY USAGE command
+* copy: Supports a deep copy of bloom filter objects and is used by the COPY command
+* free_effort: Determine whether the bloom object's memory needs to be lazy reclaimed or synchronously freed. We return
+    the number of filters in the bloom object as the free effort and this is similar to how the core handles free_effort
+    for aggregated objects.
+
+
+### Replication
+
+Every Bloom Filter based write operation (bloom object creations, scaling, setting of an item on a filter which returns 1
+indicating a new entry, etc) will be replicated to replica nodes. Attempts of adding an already existing item to a bloom
+object (which returns 0) will not replicated.
+
+### Keyspace Event Notification
+
+Every bloom filter based write command (that involves mutation as explained in the section above) will be made to publish
+a keyspace event after the data is mutated. Commands include: BF.RESERVE, BF.ADD, BD.MADD, and BF.INSERT.
+
+* Event type: VALKEYMODULE_NOTIFY_GENERIC
+* Event name: One of the two event names will be published based on the command & scenario:
+ * bloom.add: Any BF.ADD, BF.MADD, or BF.INSERT command that results in adding item/s to a bloom object.
+ * bloom.reserve: Any BF.ADD, BF.MADD, BF.INSERT, or BF.RESERVE command that results in creation of a bloom object.
+
+
+Users can subscribe to the bloom events via the standard keyspace event pub/sub. For example,
+
+```text
+1. enable keyspace event notifications:
+    valkey-cli config set notify-keyspace-events KEA
+2. suscribe to keyspace & keyevent event channels:
+    valkey-cli psubscribe '__key*__:*'
+```
```

Comment:
This is a generic mechanism for all keys. Do you need to implement anything specifically for Bloom module?

Reply:
No, there should not be anything to implement specifically. There is a generic Module keyspace notification system (and APIs) that we can utilize to publish these events.